Examples/clipboard-demo.lisp

```diff
@@ -0,0 +1,116 @@
+    (obj (type markup-text) (output-type (eql :string)) check-only)
+  (markup-text/text obj))
+
+(clim:define-presentation-method clim-extensions:convert-clipboard-content
+    (obj (type markup-text) (output-type (eql :html)) check-only)
+  (format nil "Highlighted content: <b>~a</b>" (markup-text/text obj)))
+
+(defclass paste-demo-stream-pane (clim:application-pane)
+  ())
+
+(defclass paste-output-stream-pane (clim:application-pane)
+  ())
+
+(clim:define-application-frame clipboard-demo ()
+  ((pasted-list :initform nil
+                :accessor clipboard-demo/pasted-list))
+  (:panes (content (clim:make-pane 'paste-demo-stream-pane
+                                   :display-function 'display-clipboard-demo
+                                   :scroll-bars :both
+                                   :incremental-redisplay nil
+                                   :end-of-line-action :wrap*))
+          (paste-output (clim:make-pane 'paste-output-stream-pane
+                                        :display-function 'display-paste-output
+                                        :incremental-redisplay nil
+                                        :end-of-line-action :wrap*))
+          (interaction-pane :interactor))
+  (:menu-bar clipboard-demo-menubar-table)
+  (:geometry :width 1000 :height 600)
+  (:layouts (default (clim:vertically ()
+                       (2/3 (clim:horizontally ()
+                              (clim:scrolling ()
+                                content)
+                              (clim:scrolling ()
+                                paste-output)))
+                       (1/3 interaction-pane)))))
+
+(clim:define-presentation-to-command-translator copy-markup-text
+    (markup-text climi::com-copy-to-clipboard clipboard-demo :menu t)
+    (obj)
+  (list (make-instance 'climi::clipboard-object :content obj :type 'markup-text)))
+
+(defmethod clim:handle-event :around ((pane paste-output-stream-pane) (event clim-extensions:clipboard-send-event))
+  (let ((frame (clim:pane-frame pane)))
+    (setf (clipboard-demo/pasted-list frame)
+          (append (clipboard-demo/pasted-list frame)
+                  (list (list (clim-extensions:clipboard-event-content event)
+                              (clim-extensions:clipboard-event-type event)))))
+    (clim:redisplay-frame-pane frame pane)))
+
+(defun display-clipboard-demo (frame stream)
+  (declare (ignore frame))
+  (format stream "Select text using shift and the left mouse button~%")
+  (format stream "You can also use the command \"Copy to clipboard\" to select a presentation.~%~%")
+  (let ((obj (make-instance 'markup-text :text "This text can be copied in HTML form")))
+    (clim:stream-present stream obj (clim:presentation-type-of obj)))
+  (format stream "~%~%You can paste the clipboard into editor fields using Shift-Insert,~%")
+  (format stream "or you can Shift-click the middle mouse button to paste from the selection.~%~%")
```

Comment:
shift-click-middle-button gesture doesn't work here.

Reply:
Shift-middle-click is only implemented where you can type text (designed to behave similar to GNOME applications). It is of course possible to implement it in the clipboard-demo.

Reply:
I know, I've tried to put it in the edit box.

Reply:
That's strange. It works for me. Can you provide an exact step-by-step explanation as to what you do? Perhaps there is some edge-case where the event isn't delivered in some circumstances. The more exact you can be, the better.